Doc/library/contextlib.rst

```diff
@@ -47,22 +47,28 @@ Functions and classes provided:
    function for :keyword:`with` statement context managers, without needing to
    create a class or separate :meth:`__enter__` and :meth:`__exit__` methods.
 
-   A simple example (this is not recommended as a real way of generating HTML!)::
+   While many objects natively support use in with statements, sometimes a
+   resource needs to be managed that isn't a context manager in its own right,
+   and doesn't implement a ``close()`` method for use with ``contextlib.closing``
+
+   A simple example thus would be the following to ensure correct resource
+   management::
```

Comment:
IMO the fact that this an abstract example (rather than a concrete one) should be mentioned explicitly.

Also, I suggest removing the word "simple".